src/contrib/cluster/Akka.Cluster.Metrics/Serialization/NodeMetrics.cs

```diff
@@ -10,19 +10,41 @@
 using System.Collections.Immutable;
 using System.Linq;
 using Akka.Util;
-using Google.Protobuf.Collections;
 
+#nullable enable
 namespace Akka.Cluster.Metrics.Serialization
 {
+    internal sealed class NodeMetricsComparer: IEqualityComparer<NodeMetrics>
```

Comment:
This is needed because NodeMetrics only uses the Address property for equality and hash code computation. This is being used internally by MetricGossip ImmutableHashSet property to determine if 2 instances of metrics are the same or not. ImmutableHashSet first uses the GetHashCode() method to retrieve items, if more than one items are retrieved, it will use the Equals() method to get the more accurate match.